util/len-caching-lock/src/mutex.rs

```diff
@@ -46,62 +31,78 @@ pub struct LenCachingMutex<T> {
 }
 
 impl<T: Len> LenCachingMutex<T> {
+	/// Constructs a new LenCachingMutex
 	pub fn new(data: T) -> LenCachingMutex<T> {
 		LenCachingMutex {
 			len: AtomicUsize::new(data.len()),
 			data: Mutex::new(data),
 		}
 	}
 
-	/// Load the most recent value returned from your `T`'s `len()`
+	/// Load the cached value that was returned from your `T`'s `len()`
+	/// subsequent to the most recent lock being released.
 	pub fn load_len(&self) -> usize {
 		self.len.load(Ordering::SeqCst)
 	}
 
-	pub fn lock(&self) -> Guard<T> {
-		Guard {
+	/// Convenience method to check if collection T `is_empty()`
+	pub fn load_is_empty(&self) -> bool {
```

Comment:
I don't think it's worthwhile unless it's specialised via another `AtomicBool` (which probably doesn't make much sense). I'd remove that method, doing `self.load_len() == 0` is self-explanatory.

Reply:
Sure, I was debating whether to provide it...